### ISSUE TITLE
Need a way to convert an ImageSource into an IImage.

### ISSUE THREAD

**Issue by GuildOfCalamity** [dotnet/maui#10624]
### Description

Would love to have a way  for the `ImageSource` object to be converted to `IImage` for `Canvas.DrawImage` calls on Windows environments.

### Public API Changes

```
var imgSource = ImageSource.FromStream(() => stream).ToIImage();
/* or */
var imgSource = (IImage)ImageSource.FromStream(() => stream);
```

### Intended Use-Case

The `Microsoft.Maui.Graphics.Platform.PlatformImage.FromStream(stream)` returns an `IImage` type, which is not supported on Windows :(
There needs to be a cast option or a public converter method so that images can bee used in canvas objects on Windows platforms.

**Comment by janseris** [comment]
https://github.com/dotnet/maui/issues/6742

I don't know if it was really ever solved. 
MAUI.Graphics subproject has been probably dead since then.

**Comment by PureWeen** [comment]
@mattleibow

**Comment by GuildOfCalamity** [comment]
Does anyone know how to convert between these two? I'm not against adding extra code to handle it, but I can't find any specs on the low-level object types and what their native forms are. Maybe they are `Bitmap` objects at some point?

**Comment by mattleibow** [comment]
Maybe you can get the imagesourceservice from the service provider, get the image and then do a new platformimage(img)

Converting back may be harder as the source expects a new image each time, but maybe a new imagesourceservice is something that can be added to always return the same image. 

// typing on a phone now

**Comment by philipag** [comment]
Something like this will get the image from a stream and presumably you can get the stream if you know what you image source is:

`			var assembly = GetType().GetTypeInfo().Assembly;
			using (var stream = assembly.GetManifestResourceStream("TDSP.Resources.Images.splash_light_background_no_motto.png"))
				iimage = PlatformImage.FromStream(stream);
`

**Comment by janseris** [comment]
@philipag this only applies to images in project resources provided with the application doesn't it

**Comment by philipag** [comment]
@janseris No, it applies to any stream.  So if you know the path you can just open the file as a stream:

`		using (var stream = new FileStream(photo.Path, FileMode.Open))
			image = PlatformImage.FromStream(stream);`

**Comment by janseris** [comment]
@philipag PlatformImage is Android API

**Comment by philipag** [comment]
@janseris Microsoft.Maui.Graphics.Platform.PlatformImage is part of Maui and returns a MAUI IImage.

**Comment by GuildOfCalamity** [comment]
As @janseris states, Windows build environments do not support `PlatformImage.FromStream()`

**Comment by faruknane** [comment]
@GuildOfCalamity Do you know a way to convert Image.Source (ImageSource) to IImage for other platforms?

**Comment by zerihal** [comment]
For Windows platform, you should be able to use `W2DImageLoadingService().FromStream` to convert a stream to a IImage, otherwise `PlatformImage.FromStream` for other platforms. To avoid compiler errors, you could have a helper class/method like the following to return an IImage from a stream for all platforms:

```
using IImage = Microsoft.Maui.Graphics.IImage;
#if !WINDOWS
using Microsoft.Maui.Graphics.Platform;
#else
using Microsoft.Maui.Graphics.Win2D;
#endif

namespace SomeMauiApp.Util
{
    public static class XPlatformImageHelper
    {
        /// <summary>
        /// Gets IImage from image source stream for all platforms.
        /// </summary>
        /// <param name="source">Image source stream.</param>
        /// <returns>MAUI IImage implementation from the image stream.</returns>
        public static IImage GetImageFromStream(Stream source)
        {
          #if WINDOWS
            return new W2DImageLoadingService().FromStream(source);
          #else
            return PlatformImage.FromStream(source);
          #endif
        }
    }
}
```

So using the above, if you wanted to get an image from resources and then apply to `Canvas.DrawImage`, this could be done with something like the below:

`canvas.DrawImage(XPlatformImageHelper.GetImageFromStream(new MemoryStream(Resources.someImage)), _imgX, _imgY, _imgWidth, _imgHeight);`
